src/datasources/staking-api/kiln-api.service.ts

```diff
@@ -281,15 +246,23 @@ export class KilnApi implements IStakingApi {
    */
   private async get<T>(
     args: Parameters<typeof this.dataSource.get>[0],
-  ): Promise<{ data: Raw<T> }> {
-    return this.dataSource
-      .get<{
-        data: TransactionStatus;
-      }>(args)
-      .then((res) => {
-        // Ensuring response is { data: T }, the data is parsed in the domain
-        return z.object({ data: z.unknown() }).parse(res) as { data: Raw<T> };
-      });
+  ): Promise<Raw<T>> {
+    try {
+      const { data } = await this.dataSource
+        .get<{
+          data: TransactionStatus;
+        }>(args)
+        .then((res) => {
+          // Ensuring response is { data: T }, the data is parsed in the domain
+          return z.object({ data: z.unknown() }).parse(res) as { data: Raw<T> };
+        });
+      return data;
+    } catch (error) {
+      if (error instanceof ZodError) {
+        throw error;
+      }
```

Comment:
`ZodError`s are used for types and are inherently not `DataSourceError`s.

Reply:
https://github.com/safe-global/safe-client-gateway/pull/2166#discussion_r1869786237

Reply:
See https://github.com/safe-global/safe-client-gateway/pull/2166#discussion_r1869790236